globset/src/glob.rs

```diff
@@ -194,6 +194,8 @@ struct GlobOptions {
     /// Whether to require a literal separator to match a separator in a file
     /// path. e.g., when enabled, `*` won't match `/`.
     literal_separator: bool,
+    /// doc
+    from_regex: bool,
```

Comment:
Ah, right, yeah you'll have to do something like this:

```rust
pub struct Glob(GlobInner);

enum GlobInner {
    Glob { ... },
    Regex(String),
}
```

The other interesting bit here is that by offering a constructor that accepts a regex, we've essentially tied to the implementation itself to regexes as well. It's mildly unfortunate, but I think it's the right path forward for now.